pyomo/solvers/tests/checks/test_no_solution_behavior.py

```diff
@@ -0,0 +1,110 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import os
+import types
+try:
+    import new
+    new_available=True
+except:
+    new_available=False
+
+import pyutilib.th as unittest
+
+from pyomo.solvers.tests.models.base import test_models
+from pyomo.solvers.tests.testcases import test_scenarios
+
+# The test directory
+thisDir = os.path.dirname(os.path.abspath( __file__ ))
+
+# Cleanup Expected Failure Results Files
+_cleanup_expected_failures = True
+
+#
+# A function that function that returns a function that gets
+# added to a test class.
+#
+def create_test_method(model,
+                       solver,
+                       io,
+                       test_case):
+
+    is_expected_failure = test_case.status == 'expected failure'
+
+    #
+    # Create a function that executes the test
+    #
+    def failed_solve_test(self):
+        # Create the model test class
+        model_class = test_case.model()
+
+        # Create the model instance
+        model_class.generate_model(test_case.testcase.import_suffixes)
+        model_class.warmstart_model()
+
+        # solve
+        load_solutions = True
+        symbolic_labels = False
+        opt, results = model_class.solve(
+            solver,
+            io,
+            test_case.testcase.io_options,
+            test_case.testcase.options,
+            symbolic_labels,
+            load_solutions)
```

Comment:
This "test" worries me, as it does not assert anything.  Can you wrap this in a LoggingIntercept context manager to ensure that the expected error is emitted?

Reply:
The problem is that I've only guaranteed logging when no solution exists for Kernel models. The AML logging output happens in `model.solutions.load_from` based on the status.

If someone want to expand the scope of the PR to make changes to the solver for AML models, I am okay with that.

Reply:
Given that this is an error and not an exception, I am OK with making the change for the AML side, too.

Reply:
(Actually, what I meant to say is I am strongly in favor of adding it to the AML side)